vault/audit_broker.go

```diff
@@ -227,7 +208,12 @@ func (a *AuditBroker) GetHash(ctx context.Context, name string, input string) (s
 
 // LogRequest is used to ensure all the audit backends have an opportunity to
 // log the given request and that *at least one* succeeds.
-func (a *AuditBroker) LogRequest(ctx context.Context, in *logical.LogInput, headersConfig *AuditedHeadersConfig) (ret error) {
```

Comment:
Another curiosity question: did we remove the AuditedHeadersConfig functionality when moving to the eventlogger? It seems curious that we only supported it in the legacy mode (and now, not at all).

Reply:
Totally valid question 👍🏼 

The audit headers config is still supplied, *but* it happens during creation of the audit backend rather than per `LogRequest`/`LogResponse` call. 

The `Factory` methods all accept an `audit.HeaderFormatter` interface which will be a pointer to the instance of the `AuditedHeadersConfig` which lives in a field on `Core`, is assigned in `Core.setupAuditedHeadersConfig` and gets updated there when folks add headers they want to appear in the audit log.

Reply:
Got it! I figured this was the case (it makes more sense to be this way) but i couldn't spot it when I sniffed around. Thanks!